### ISSUE TITLE
Add basic documentation

### ISSUE THREAD

**Issue by mirenradia** [GRTLCollaboration/GRTeclyn#17]
Consider switching to [Sphinx](https://docs.readthedocs.io/en/stable/intro/getting-started-with-sphinx.html) rather than using the GitHub wiki.

**Comment by KAClough** [comment]
I have considered this, and I would be happy to stick with the GitHub wiki, which has so far worked fine for us :-)

**Comment by KAClough** [comment]
@mirenradia makes a good point about being able to update the wiki when doing a PR for an update to the code. A solution that is suggested is to add the GitHub wiki as a submodule of the code... but it seems a bit complicated...

**Comment by mirenradia** [comment]
I don't think that would work very well. I think it's best for the documentation source files (e.g. markdown files) to just reside in the code repository (e.g. in a docs subdirectory) and then it can be built using CI/CD.

**Comment by KAClough** [comment]
I would like to keep the ability to just modify the docs using the GUI in GitHub though. Ideally the system could support both. Making it something that requires CI/a commit and PR every time creates a barrier to people (me) making small but useful amendments.

**Comment by mirenradia** [comment]
You can edit Markdown files on GitHub but yes you would need to open a PR to make a change. I agree that it is a little bit more of a barrier but I don't think this is a bad thing for the following reasons:

1. If it's just a minor update, you wouldn't need to put much effort into the PR description or title. It's only a minor barrier.
2. A small update to the documentation PR can be an easy way for a new user to get into the habit of contributing to the code.
3. Since PRs are reviewed, the quality of the documentation may be improved.

**Comment by KAClough** [comment]
This has been started in the wiki here, but we have decided to move it to the read the docs format but using markdown.

**Comment by julianakwan** [comment]
We have a [documentation page](https://grtlcollaboration.github.io/GRTeclyn/) now! I've made a start on generating online documentation with `MkDocs`. The markdown files are kept in the [enhancement/documentation](https://github.com/GRTLCollaboration/GRTeclyn/tree/enhancement/documentation) branch in the `./docs` directory and I build to the `gh-pages` branch. I've moved (some of) the wiki pages over but it is still very much a work in progress. NB: I've had to change the branch where GitHub pages loads the site information from `develop` to `gh-pages`.

Please feel free to edit or add to the pages. Building the docs is done manually at the moment (but I will add it to the CI/CD at some point). Here is a quick guide on how: 

0. Check out the `enhancement/documentation` branch. 
1.  Install MkDocs using `pip install --user mkdocs`. I do this in a virtual environment so as not to pollute my python packages, in which case `--user` can be dropped. 
2. Edit the `*.mk` files in `/docs`. New pages can be added by editing `mkdocs.yml` under `nav:`. `about.mk` is there as a style guide for markdown, I will remove it later. 
3. (optional) Preview your changes locally using `mkdocs serve`. It will give you a local URL that you can paste into your browser. Changes in the `.mk` files and `mkdocs.yml` will automatically appear. 
4. Build and deploy the docs to the GRTeclyn website: 
```
mkdocs gh-deploy --config-file ../mkdocs.yml --remote-branch=gh-pages
```

I had a lot of trouble getting it to work with GitHub Pages (as you can see in the git history 😬 ) so please let me know if anything is unclear. TLDR: edit `enhancement/documentation` and build in `gh-pages` 🥹

**Comment by mirenradia** [comment]
Thanks @julianakwan for setting this up. It looks great!

I presume you intend to add a CI workflow to build and deploy it before we merge this into `develop` so that editors don't have to manually deploy?

Another alternative to installing `mkdocs` with `pip install --user` or in a manually created virtual environment is to install with [`pipx`](https://pipx.pypa.io/stable/) (which automatically creates a virtual environment for the package).

I presume you have already noticed lots of things that need to be fixed but one thing I noticed is that the links of the form `[[ xxx | yyy ]]` are broken. I have actually never seen this syntax outside of the GRTLCollaboration wikis (and it's not in the GitHub Flavoured Markdown documentation) so I think it's non-standard.

**Comment by julianakwan** [comment]
Just letting you know that I have implemented a GitHub workflow that builds and deploys our `MkDocs` pages automatically on push. 

(I wish the other workflows didn't run as well but oh well). 

I tested it by fixing some of the other broken links 😅

**Comment by mirenradia** [comment]
> Just letting you know that I have implemented a GitHub workflow that builds and deploys our MkDocs pages automatically on push.

Great 😄 Thanks for setting this up!

> (I wish the other workflows didn't run as well but oh well).

We should investigate using [path filters](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#onpushpull_requestpull_request_targetpathspaths-ignore) to control when workflows run.

**Comment by KAClough** [comment]
We should now update the README and make the link to the docs clear.
